olive/passes/onnx/conversion.py

```diff
@@ -195,22 +219,95 @@ def _convert_model_on_device(
+        os.environ["OMPI_MCA_pml"] = "ob1"  # noqa: SIM112
+        os.environ["OMPI_MCA_btl_tcp_if_include"] = "eth0"  # noqa: SIM112
+        os.environ["OMPI_MCA_hwloc_base_binding_policy"] = "numa"  # noqa: SIM112
+        os.environ["OMPI_MCA_ess"] = "^singleton"  # noqa: SIM112
+        os.environ["OMPI_MCA_ess_base_vpid"] = "0"  # noqa: SIM112
+        os.environ["OMPI_MCA_orte_tag_output"] = "1"  # noqa: SIM112
+        os.environ["OMPI_MCA_pmix"] = "^s1,s2,cray,isolated"  # noqa: SIM112
+        os.environ["OMPI_MCA_rmaps_ppr_n_pernode"] = "1"  # noqa: SIM112
+        os.environ["NCCL_DEBUG"] = "WARN"
+
+        from mpi4py.MPI import COMM_WORLD
+
+        world_size = COMM_WORLD.Get_size()
+        local_rank = COMM_WORLD.Get_rank()
+
+        torch.distributed.init_process_group(
+            "nccl", init_method="tcp://127.0.0.1:9876", world_size=world_size, rank=local_rank
+        )
+
+        output_filename = DistributedOnnxModel.DEFAULT_RANKED_MODEL_NAME_FORMAT.format(local_rank)
+        output_filepath = str(Path(output_dirpath) / output_filename)
+
+        input_model = DistributedPyTorchModel(**model_config)
+        pytorch_model = input_model.load_model(local_rank)
+
+        dummy_inputs = pytorch_model.get_dummy_inputs()
+        io_config = None if pass_config["use_dynamo_exporter"] else get_io_config(pytorch_model)
+        pytorch_model = pytorch_model.prepare_session(rank=local_rank)
+
+        COMM_WORLD.Barrier()
+        ranked_onnx_model = OnnxConversion._convert_model_on_device(
+            pytorch_model, dummy_inputs, io_config, data_root, pass_config, output_filepath, device
+        )
+        COMM_WORLD.Barrier()
 
-    def _run_for_config(
-        self, model: PyTorchModel, data_root: str, config: Dict[str, Any], output_model_path: str
-    ) -> Union[ONNXModel, CompositeOnnxModel]:
-        accel_type = self.accelerator_spec.accelerator_type
-        device = torch.device("cuda") if accel_type == Device.GPU else torch.device(accel_type)
-        return self._convert_model_on_device(model, data_root, config, output_model_path, device)
+        # save the model to the output path and return the model
+        model_proto_to_olive_model(ranked_onnx_model, output_filepath, pass_config)
+
+    def _convert_distributed_model_on_device(
+        self,
+        model: DistributedPyTorchModel,
+        data_root: str,
+        config: Dict[str, Any],
+        output_model_path: str,
+        device: str,
+    ) -> DistributedOnnxModel:
+        from mpi4py.futures import MPIPoolExecutor
+
+        pass_config = config
+        model_config = model.to_json()["config"]
+        world_size = model.num_ranks
+
+        params = [
+            (
+                pass_config,
+                data_root,
+                model_config,
+                world_size,
+                device,
+                rank,
+                output_model_path,
```

Comment:
nit: the `output_model_path` could be resolved beforehand as 
`output_model_path = str(Path(output_model_path).with_suffix(""))` 

and use here and in the return part to be sure we are using the same value in both.

Reply:
Done!